### PR TITLE
web-nfc: add IDL harness tests

### PR DESCRIPTION
**Pull request by Honry**
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/5267)

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/216660143)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 65463242c65e06dbf4388b7ad2a13d7b74614e57
Using browser at version BuildID 20170329100319; SourceStamp 272ce6c2572164f5f6a9fba2a980ba9ccf50770c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="https://w3c-test.org/submissions/5267/web-nfc/idlharness.https.html">/web-nfc/idlharness.https.html</a></summary>


|                                                           Subtest                                                            | Results |                                           Messages                                           |
|------------------------------------------------------------------------------------------------------------------------------|---------|----------------------------------------------------------------------------------------------|
|                                                                                                                              | OK      |                                                                                              |
| `Navigator interface: attribute nfc`                                                                                         | FAIL    | `assert_true: The prototype object must have a property "nfc" expected true got false`       |
| `NFC interface: existence and properties of interface object`                                                                | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface object length`                                                                                                | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface object name`                                                                                                  | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: existence and properties of interface prototype object`                                                      | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: existence and properties of interface prototype object's "constructor" property`                             | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: operation push(NFCPushMessage,NFCPushOptions)`                                                               | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: operation cancelPush(NFCPushTarget)`                                                                         | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: operation watch(MessageCallback,NFCWatchOptions)`                                                            | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: operation cancelWatch(long)`                                                                                 | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC must be primary interface of navigator.nfc`                                                                             | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `Stringification of navigator.nfc`                                                                                           | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: navigator.nfc must inherit property "push" with the proper type (0)`                                         | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: calling push(NFCPushMessage,NFCPushOptions) on navigator.nfc with too few arguments must throw TypeError`    | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: navigator.nfc must inherit property "cancelPush" with the proper type (1)`                                   | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: calling cancelPush(NFCPushTarget) on navigator.nfc with too few arguments must throw TypeError`              | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: navigator.nfc must inherit property "watch" with the proper type (2)`                                        | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: calling watch(MessageCallback,NFCWatchOptions) on navigator.nfc with too few arguments must throw TypeError` | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: navigator.nfc must inherit property "cancelWatch" with the proper type (3)`                                  | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: calling cancelWatch(long) on navigator.nfc with too few arguments must throw TypeError`                      | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/216660144)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 65463242c65e06dbf4388b7ad2a13d7b74614e57
Using browser at version 59.0.3053.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="https://w3c-test.org/submissions/5267/web-nfc/idlharness.https.html">/web-nfc/idlharness.https.html</a></summary>


|                                                           Subtest                                                            | Results |                                           Messages                                           |
|------------------------------------------------------------------------------------------------------------------------------|---------|----------------------------------------------------------------------------------------------|
|                                                                                                                              | OK      |                                                                                              |
| `Navigator interface: attribute nfc`                                                                                         | FAIL    | `assert_true: The prototype object must have a property "nfc" expected true got false`       |
| `NFC interface: existence and properties of interface object`                                                                | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface object length`                                                                                                | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface object name`                                                                                                  | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: existence and properties of interface prototype object`                                                      | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: existence and properties of interface prototype object's "constructor" property`                             | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: operation push(NFCPushMessage,NFCPushOptions)`                                                               | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: operation cancelPush(NFCPushTarget)`                                                                         | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: operation watch(MessageCallback,NFCWatchOptions)`                                                            | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC interface: operation cancelWatch(long)`                                                                                 | FAIL    | `assert_own_property: self does not have own property "NFC" expected property "NFC" missing` |
| `NFC must be primary interface of navigator.nfc`                                                                             | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `Stringification of navigator.nfc`                                                                                           | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: navigator.nfc must inherit property "push" with the proper type (0)`                                         | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: calling push(NFCPushMessage,NFCPushOptions) on navigator.nfc with too few arguments must throw TypeError`    | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: navigator.nfc must inherit property "cancelPush" with the proper type (1)`                                   | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: calling cancelPush(NFCPushTarget) on navigator.nfc with too few arguments must throw TypeError`              | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: navigator.nfc must inherit property "watch" with the proper type (2)`                                        | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: calling watch(MessageCallback,NFCWatchOptions) on navigator.nfc with too few arguments must throw TypeError` | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: navigator.nfc must inherit property "cancelWatch" with the proper type (3)`                                  | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |
| `NFC interface: calling cancelWatch(long) on navigator.nfc with too few arguments must throw TypeError`                      | FAIL    | `assert_equals: wrong typeof object expected "object" but got "undefined"`                   |

</details>
</details>

**Comment by Honry** [comment]
@kenchris or @zolkis,  could you help review this PR?

File: web-nfc/idlharness.https.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Web-NFC IDL tests</title>
```
Comment (kenchris) [inline web-nfc/idlharness.https.html]: no -

Web NFC

File: web-nfc/idlharness.https.html [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    NFCRecordType? recordType;
+    USVString      mediaType = "";
+    NFCWatchMode   mode = "web-nfc-only";
+};
+</pre>
+
+<script>
+setup(() => {
+   "use strict";
+    var idl_array = new IdlArray();
```
Comment (kenchris) [inline web-nfc/idlharness.https.html]: Can we use ES2015, then const instead of var

idlArray

File: web-nfc/idlharness.https.html [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    idl_array.add_untested_idls(document.getElementById('idl').textContent);
+    idl_array.add_idls(document.getElementById('web-nfc').textContent);
+    idl_array.add_objects({NFC: ["navigator.nfc"]});
+    idl_array.test();
+    done();
+}, {explicit_done: true});
```
Comment (kenchris) [inline web-nfc/idlharness.https.html]: space after { and before }

lgtm

**Comment by zolkis** [comment]
Thanks. lgtm.

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5267/)

**Comment by Honry** [comment]
@kenchris, many thanks for the comments, I just refresh the PR, pls. review again.

**Comment by Honry** [comment]
@dontcallmedom, could you help merge this PR? I think the CI failure is nothing to do with the tests.

**Pull request merged**